src/lemmas.k.md

```diff
@@ -230,31 +253,14 @@ rule notMaxUInt112xPow112 => 115792089210356248756420345214020892766250359184300
 Mask target range to zero:
 
 ```k
-// all regions potentially non-zero
-rule (notMaxUInt112xPow112 &Int ((Z *Int pow224) +Int ((Y *Int pow112) +Int X))) => ((Z *Int pow224) +Int X)
-  requires #rangeUInt(112, X)
-  andBool #rangeUInt(112, Y)
-  andBool #rangeUInt(32, Z)
-
-// reserve1 is zero
-rule (notMaxUInt112xPow112 &Int ((Z *Int pow224) +Int X)) => ((Z *Int pow224) +Int X)
-  requires #rangeUInt(112, X)
-  andBool #rangeUInt(32, Z)
+rule notMaxUInt112xPow112 &Int #WordPackUInt112UInt112UInt32(A, B, C) => #WordPackUInt112UInt112UInt32(A, 0, C)
 ```
 
 Write new value:
 
 ```k
-// all regions potentially non-zero OR reserve1 is zero
-rule ((pow112 *Int Y) |Int ((Z *Int pow224) +Int X)) => ((Z *Int pow224) +Int (Y *Int pow112) +Int X)
+rule (pow112 *Int X) |Int #WordPackUInt112UInt112UInt32(A, 0, C) => #WordPackUInt112UInt112UInt32(A, X, C)
```

Comment:
there could be code where X *Int pow112 is the case

Reply:
True. We should normalize this too I guess.

Reply:
I'm  going to leave it as is, if it breaks something we'll add the lemma with the terms reversed.